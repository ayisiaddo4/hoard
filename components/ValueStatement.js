import React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';

export default function ValueStatement({
  title,
  value,
  change,
  positive,
}) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.value}>{value}</Text>
      <Text style={[ styles.change, positive ? styles.changePositive : styles.changeNegative ]}>
        {change}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  title: {
    color: 'lightblue',
  },
  value: {
    fontSize: 30,
    color: 'white',
  },
  change: {},
  changePositive: {
    color: 'green',
  },
  changeNegative: {
    color: 'red',
  },
});
