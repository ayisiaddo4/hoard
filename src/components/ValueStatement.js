import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text } from 'react-native';
import { getColors } from 'styles';

const ValueStatement = ({ title, value, change, positive }) => {
  const themedStyles = getThemedStyles(getColors());
  return (
    <View style={styles.container}>
      <Text style={themedStyles.title}>{title}</Text>
      <Text style={[styles.value, themedStyles.value]}>{value}</Text>
      <Text
        style={[
          positive ? themedStyles.changePositive : themedStyles.changeNegative,
        ]}
      >
        {change}
      </Text>
    </View>
  );
};

ValueStatement.propTypes = {
  change: PropTypes.string.isRequired,
  positive: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  value: {
    fontSize: 30,
  },
});

const getThemedStyles = colors => {
  return {
    title: {
      color: colors.textSecondary,
    },
    value: {
      color: colors.textPrimary,
    },
    changePositive: {
      color: colors.textPositive,
    },
    changeNegative: {
      color: colors.textNegative,
    },
  };
};

export default ValueStatement;
