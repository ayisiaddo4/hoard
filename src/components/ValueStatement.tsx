import * as React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';
import {getColors, Colors} from 'styles';

interface Props {
  change: string;
  positive: boolean;
  title: string;
  value: string;
}

const ValueStatement: React.SFC<Props> = ({
  title,
  value,
  change,
  positive,
})  => {
  const themedStyles = getThemedStyles(getColors());
  return (
    <View style={styles.container}>
      <Text style={themedStyles.title}>{title}</Text>
      <Text style={[styles.value, themedStyles.value]}>{value}</Text>
      <Text style={[positive ? themedStyles.changePositive : themedStyles.changeNegative ]}>
        {change}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center'
  },
  value: {
    fontSize: 30,
  },
});

const getThemedStyles = (colors: Colors) => {
  return {
    title: {
      color: colors.textSecondary,
    },
    value: {
      color: colors.textPrimary,
    },
    changePositive: {
      color: colors.textPositive,
    },
    changeNegative: {
      color: colors.textNegative,
    },
  };
}

export default ValueStatement;
