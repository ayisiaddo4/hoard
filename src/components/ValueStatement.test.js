/* eslint-env jest */

import React from 'react';
import ValueStatement from './ValueStatement';

import renderer from 'react-test-renderer';

const MockedComponent = () => (
  <ValueStatement change="Change" positive={true} title="Title" value="Value" />
);

describe('<ValueStatement />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MockedComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// it('componentWillMount: should set the passed initialCountValue to state', () => {
//   const wrapper = shallow(<Counter initialCountValue={2} />);
//   expect(wrapper.instance().state.count).toBe(2);
// });
