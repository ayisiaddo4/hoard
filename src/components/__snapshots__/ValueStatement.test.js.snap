// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValueStatement /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#17181e",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "letterSpacing": 0,
        },
        Object {
          "color": "#8D9FAE",
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "HelveticaNeue",
          "fontSize": 32,
          "letterSpacing": 0,
        },
        Array [
          Object {
            "fontSize": 30,
          },
          Object {
            "color": "#FFFFFF",
          },
        ],
      ]
    }
  >
    Value
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00EC9E",
          "fontFamily": "HelveticaNeue",
          "fontSize": 12,
          "letterSpacing": 1,
        },
        Array [
          Object {
            "color": "#1EDC6B",
          },
        ],
      ]
    }
  >
    Change
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#17181e",
          "fontFamily": "HelveticaNeue",
          "fontSize": 14,
          "letterSpacing": 0,
        },
        Object {
          "color": "#8D9FAE",
        },
      ]
    }
  >
    Title
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "HelveticaNeue",
          "fontSize": 32,
          "letterSpacing": 0,
        },
        Array [
          Object {
            "fontSize": 30,
          },
          Object {
            "color": "#FFFFFF",
          },
        ],
      ]
    }
  >
    Value
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#00EC9E",
          "fontFamily": "HelveticaNeue",
          "fontSize": 12,
          "letterSpacing": 1,
        },
        Array [
          Object {
            "color": "#1EDC6B",
          },
        ],
      ]
    }
  >
    Change
  </Text>
</View>
`;
